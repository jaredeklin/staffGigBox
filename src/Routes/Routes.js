import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import ScheduleContainer from '../ScheduleContainer/ScheduleContainer';
import { Availability } from '../Availability/Availability';
import { StaffForm } from '../StaffForm/StaffForm';
import { EventForm } from '../EventForm/EventForm';

export const Routes = props => {
  const { currentUser, addEvent, addStaff, admin } = props;

  return (
    <Switch>
      {/* <Route exact path="/" /> */}
      <Route
        exact
        path="/schedule"
        render={() => (
          <ScheduleContainer methods={methods} appState={appState} />
        )}
      />
      <Route
        path="/schedule/ogden"
        render={() => (
          <ScheduleContainer methods={methods} appState={appState} />
        )}
      />
      <Route
        path="/schedule/gothic"
        render={() => (
          <ScheduleContainer methods={methods} appState={appState} />
        )}
      />
      <Route
        path="/schedule/bluebird"
        render={() => (
          <ScheduleContainer methods={methods} appState={appState} />
        )}
      />
      <Route
        path="/schedule/individual"
        render={() => (
          <ScheduleContainer methods={methods} appState={appState} />
        )}
      />
      <Route
        path="/unscheduled-events"
        render={() => (
          <ScheduleContainer methods={methods} appState={appState} />
        )}
      />
      {appState.currentUser.staff_id && (
        <Route
          path="/availability"
          render={() => (
            <Availability currentUserId={appState.currentUser.staff_id} />
          )}
        />
      )}
      {appState.currentUser.google_id && (
        <Route
          path="/add-staff"
          render={() => (
            <StaffForm
              addStaff={addStaff}
              id={appState.currentUser.google_id}
            />
          )}
        />
      )}
      <Route
        path="/add-events"
        render={() => <EventForm addEvent={addEvent} />}
      />
      <Route render={() => <Redirect to="/schedule" />} />
    </Switch>
  );
};

export default Routes;
