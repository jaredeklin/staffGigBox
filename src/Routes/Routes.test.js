import React from 'react';
import { shallow } from 'enzyme';
import { Routes } from './Routes';

describe('Routes', () => {
  it('should match the snapshot', () => {
    const mockUser = {
      staff_id: 1,
      google_id: '34567G'
    };
    const wrapper = shallow(<Routes currentUser={mockUser} />);
    expect(wrapper).toMatchSnapshot();
  });
});
