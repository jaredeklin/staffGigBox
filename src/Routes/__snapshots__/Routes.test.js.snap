// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    currentUser={
        Object {
            "google_id": "34567G",
            "staff_id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <PropsRoute
          component={[Function]}
          currentUser={
                    Object {
                              "google_id": "34567G",
                              "staff_id": 1,
                            }
          }
          exact={true}
          path="/schedule"
/>,
        <PrivateRoute
          authed={1}
          component={[Function]}
          currentUser={
                    Object {
                              "google_id": "34567G",
                              "staff_id": 1,
                            }
          }
          path="/schedule/ogden"
          redirectTo="/"
/>,
        <PrivateRoute
          authed={1}
          component={[Function]}
          currentUser={
                    Object {
                              "google_id": "34567G",
                              "staff_id": 1,
                            }
          }
          path="/schedule/gothic"
          redirectTo="/"
/>,
        <PrivateRoute
          authed={1}
          component={[Function]}
          currentUser={
                    Object {
                              "google_id": "34567G",
                              "staff_id": 1,
                            }
          }
          path="/schedule/bluebird"
          redirectTo="/"
/>,
        <PrivateRoute
          authed={1}
          component={[Function]}
          currentUser={
                    Object {
                              "google_id": "34567G",
                              "staff_id": 1,
                            }
          }
          path="/schedule/individual"
          redirectTo="/"
/>,
        <PrivateRoute
          authed={undefined}
          component={[Function]}
          currentUser={
                    Object {
                              "google_id": "34567G",
                              "staff_id": 1,
                            }
          }
          path="/unscheduled-events"
          redirectTo="/"
/>,
        <PrivateRoute
          authed={undefined}
          component={[Function]}
          currentUserId={1}
          path="/availability"
          redirectTo="/"
/>,
        <PrivateRoute
          addStaff={undefined}
          authed={undefined}
          component={[Function]}
          id="34567G"
          path="/add-staff"
          redirectTo="/"
/>,
        <PrivateRoute
          addEvent={undefined}
          authed={undefined}
          component={[Function]}
          path="/add-events"
          redirectTo="/"
/>,
        <Redirect
          push={false}
          to="/schedule"
/>,
        " />",
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "component": [Function],
          "currentUser": Object {
            "google_id": "34567G",
            "staff_id": 1,
          },
          "exact": true,
          "path": "/schedule",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authed": 1,
          "component": [Function],
          "currentUser": Object {
            "google_id": "34567G",
            "staff_id": 1,
          },
          "path": "/schedule/ogden",
          "redirectTo": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authed": 1,
          "component": [Function],
          "currentUser": Object {
            "google_id": "34567G",
            "staff_id": 1,
          },
          "path": "/schedule/gothic",
          "redirectTo": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authed": 1,
          "component": [Function],
          "currentUser": Object {
            "google_id": "34567G",
            "staff_id": 1,
          },
          "path": "/schedule/bluebird",
          "redirectTo": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authed": 1,
          "component": [Function],
          "currentUser": Object {
            "google_id": "34567G",
            "staff_id": 1,
          },
          "path": "/schedule/individual",
          "redirectTo": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authed": undefined,
          "component": [Function],
          "currentUser": Object {
            "google_id": "34567G",
            "staff_id": 1,
          },
          "path": "/unscheduled-events",
          "redirectTo": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "authed": undefined,
          "component": [Function],
          "currentUserId": 1,
          "path": "/availability",
          "redirectTo": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addStaff": undefined,
          "authed": undefined,
          "component": [Function],
          "id": "34567G",
          "path": "/add-staff",
          "redirectTo": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "addEvent": undefined,
          "authed": undefined,
          "component": [Function],
          "path": "/add-events",
          "redirectTo": "/",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "push": false,
          "to": "/schedule",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      " />",
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <PropsRoute
            component={[Function]}
            currentUser={
                        Object {
                                    "google_id": "34567G",
                                    "staff_id": 1,
                                  }
            }
            exact={true}
            path="/schedule"
/>,
          <PrivateRoute
            authed={1}
            component={[Function]}
            currentUser={
                        Object {
                                    "google_id": "34567G",
                                    "staff_id": 1,
                                  }
            }
            path="/schedule/ogden"
            redirectTo="/"
/>,
          <PrivateRoute
            authed={1}
            component={[Function]}
            currentUser={
                        Object {
                                    "google_id": "34567G",
                                    "staff_id": 1,
                                  }
            }
            path="/schedule/gothic"
            redirectTo="/"
/>,
          <PrivateRoute
            authed={1}
            component={[Function]}
            currentUser={
                        Object {
                                    "google_id": "34567G",
                                    "staff_id": 1,
                                  }
            }
            path="/schedule/bluebird"
            redirectTo="/"
/>,
          <PrivateRoute
            authed={1}
            component={[Function]}
            currentUser={
                        Object {
                                    "google_id": "34567G",
                                    "staff_id": 1,
                                  }
            }
            path="/schedule/individual"
            redirectTo="/"
/>,
          <PrivateRoute
            authed={undefined}
            component={[Function]}
            currentUser={
                        Object {
                                    "google_id": "34567G",
                                    "staff_id": 1,
                                  }
            }
            path="/unscheduled-events"
            redirectTo="/"
/>,
          <PrivateRoute
            authed={undefined}
            component={[Function]}
            currentUserId={1}
            path="/availability"
            redirectTo="/"
/>,
          <PrivateRoute
            addStaff={undefined}
            authed={undefined}
            component={[Function]}
            id="34567G"
            path="/add-staff"
            redirectTo="/"
/>,
          <PrivateRoute
            addEvent={undefined}
            authed={undefined}
            component={[Function]}
            path="/add-events"
            redirectTo="/"
/>,
          <Redirect
            push={false}
            to="/schedule"
/>,
          " />",
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "component": [Function],
            "currentUser": Object {
              "google_id": "34567G",
              "staff_id": 1,
            },
            "exact": true,
            "path": "/schedule",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authed": 1,
            "component": [Function],
            "currentUser": Object {
              "google_id": "34567G",
              "staff_id": 1,
            },
            "path": "/schedule/ogden",
            "redirectTo": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authed": 1,
            "component": [Function],
            "currentUser": Object {
              "google_id": "34567G",
              "staff_id": 1,
            },
            "path": "/schedule/gothic",
            "redirectTo": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authed": 1,
            "component": [Function],
            "currentUser": Object {
              "google_id": "34567G",
              "staff_id": 1,
            },
            "path": "/schedule/bluebird",
            "redirectTo": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authed": 1,
            "component": [Function],
            "currentUser": Object {
              "google_id": "34567G",
              "staff_id": 1,
            },
            "path": "/schedule/individual",
            "redirectTo": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authed": undefined,
            "component": [Function],
            "currentUser": Object {
              "google_id": "34567G",
              "staff_id": 1,
            },
            "path": "/unscheduled-events",
            "redirectTo": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "authed": undefined,
            "component": [Function],
            "currentUserId": 1,
            "path": "/availability",
            "redirectTo": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addStaff": undefined,
            "authed": undefined,
            "component": [Function],
            "id": "34567G",
            "path": "/add-staff",
            "redirectTo": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "addEvent": undefined,
            "authed": undefined,
            "component": [Function],
            "path": "/add-events",
            "redirectTo": "/",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "push": false,
            "to": "/schedule",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        " />",
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
